nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/main/java/org/nuxeo/ecm/directory/PasswordHelper.java

```diff
@@ -139,7 +139,8 @@ public static boolean verifyPassword(String password, String hashedPassword) {
     public static byte[] digestWithSalt(String password, byte[] salt, String algorithm) {
         try {
             MessageDigest md = MessageDigest.getInstance(algorithm);
-            md.update(password.getBytes("UTF-8"));
+            String tempPassword = password == null ? "" : password;
```

Comment:
No need for another variable, you can just change `password` as it's not used anywhere else. We don't have rules about not changing function parameters. Just do:

    if (password == null) {
        password = "";
    }